stdlib/public/SDK/Foundation/NSStringAPI.swift

```diff
@@ -417,6 +417,20 @@ extension StringProtocol where Index == String.Index {
     return self._ephemeralString as NSString
   }
 
+  func _withCore<Result>(_ apply: (_StringCore) throws -> Result) rethrows -> Result {
```

Comment:
I have found that to be completely unpredictable.  In one place I eliminate closures and there's a big improvement.  In another, there's a regression.  Part of it might be that I used `defer { _fixLifetime(x) }` and, as I've just discovered, `defer` is implemented under the hood with closures.